pkg/storage/replica_raftstorage.go

```diff
@@ -350,6 +361,7 @@ func (r *Replica) GetSnapshot(ctx context.Context) (*OutgoingSnapshot, error) {
 			return snap, err
 		}
 	}
+	return nil, errors.New("retries exhausted")
```

Comment:
this won't ever be reached, but it does raise the question: should we set a max retries? or keep going forever?
In the latter case, I think we need a better way of finding stuck ops, but that's rather separate from this issue